### PR TITLE
Split too large txs

### LINKED ISSUE #105: Handle the `ExhaustsResources` issue
When doing a too big batch, we may get the error [`ExhaustsResources`](https://paritytech.github.io/polkadot-sdk/master/sp_runtime/transaction_validity/enum.InvalidTransaction.html#variant.ExhaustsResources)
> The transaction would exhaust the resources of current block.
The transaction might be valid, but there are not enough resources left in the current block.

I came acros this when
- undelegating 10+ tracks
- remove vote
- delegating 10+ tracks

So I imagine this may happen on Polkadot or Kusama. We need to check.

To prevent this, we need to split in different Txs, I don't think there's another way... Pinging the papi team here: https://github.com/polkadot-api/polkadot-api/issues/695

edit: follow the link, there's an answer and some code even :heart:

### PR DESCRIPTION
**Pull request by Tbaut**
closes #105
We split into first: undelegate and unvote, second: delegate.
If users want to unvote on many many refs, the first tx may also be too large, in which case it'd need to be splited too (manually to start with?). The chances that this happens are so low, that I went for a fix split into 2, and see if anyone ever opens an issue

image

---

Submission checklist:

#### Layout

- [x] Change inspected in the mobile web ui
- [x] Change inspected in dark mode

### CODE REVIEW

**Comment by wirednkod** [comment]
Something is strange here.. whatever transaction I am trying to do falls back to the TX too large, message as seen above.
Maybe I am misunderstanding the process... but it seems buggy to me 🤔

File: src/pages/Delegate/MultiTransactionDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      ].map((tx) => tx.decodedCall),
+    })
+
+    if (!step1Txs) return
+
+    // check if we have an exhausted limit on the 1st tx
+    const isExhaustsRessouces = await isExhaustsResources(step1Txs)
+
+    // this is too big of a batch we need to split it up more
+    if (!isExhaustsRessouces) {
```
Comment (Tbaut) [inline src/pages/Delegate/MultiTransactionDialog.tsx:60]:
```suggestion
    if (isExhaustsRessouces) {
```

**Comment by Tbaut** [comment]
@wirednkod sorry I left the flag to test it.. It's now removed

**Review by wirednkod — APPROVED** [review]
Beside the comment nit, I do not like the fact that there is no error handling in the requests.
I would suggest though to pass this for now, and we can tackle the error handling and reactions on a later stage. WDYT?

File: src/hooks/useTestTx.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      // {
+      //  success: false,
+      //  value: {
+      //    type: "Invalid",
+      //    value: {
+      //      type: "ExhaustsResources",
+      //      value: undefined,
+      //    },
+      //  },
+      // }
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply src/hooks/useTestTx.tsx:49]: I think it helps understand this very weird `dryRunResult.value.value.type` after, so I'd say we can keep it?

[the remark this answers is not in the document]
Comment (wirednkod) [reply src/hooks/useTestTx.tsx:49]: Sure thing

**Comment by Tbaut** [comment]
Thanks, yeah let's merge this monster, I'll refactor the error stuff once the toasts are in

**Pull request merged**